Setup build system

igchor opened the pull request:
This PR creates cmake based build system and adapts majority of cpp tests.

marcinslusarz commented:
Review status: 109 of 238 files reviewed at latest revision, all discussions resolved.

---

*[CMakeLists.txt, line 33 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfCnqx8dbswVoyLsno:-LDfCnqx8dbswVoyLsnp:b32vq9) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/CMakeLists.txt#L33)):*
> ```CMake
> 
> cmake_minimum_required(VERSION 3.3)
> project(libpmemobj-cpp C CXX)
> ```

C?

---

*[CMakeLists.txt, line 87 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfCqPhBB2Wo1_2o_VS:-LDfCqPhBB2Wo1_2o_VT:b-m27nif) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/CMakeLists.txt#L87)):*
> ```CMake
> endif()
> 
> add_definitions(-DSRCVERSION="${SRCVERSION}")
> ```

is this needed?

---

*[CMakeLists.txt, line 138 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfDWTS4um6RIunj_kW:-LDfDWTS4um6RIunj_kX:bxtub52) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/CMakeLists.txt#L138)):*
> ```CMake
> add_flag("-U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2" RELEASE)
> 
> add_flag(-D_LARGEFILE64_SOURCE)
> ```

Is it needed?

---

*[tests/CMakeLists.txt, line 102 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfFfXp1WdRwU3Q5gK_:-LDfFfXp1WdRwU3Q5gKa:b-qt029r) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/CMakeLists.txt#L102)):*
> ```CMake
> endfunction()
> 
> build_test(file_allocator allocator/allocator.cpp)
> ```

file_?

---

*[tests/CMakeLists.txt, line 113 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfFhfb0P8aDxrADDIU:-LDfFhfd454J1vbz4AyD:byvb6zf) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/CMakeLists.txt#L113)):*
> ```CMake
> build_test(file_transaction transaction/transaction.cpp)
> 
> #XXX:
> ```

Why? (please add comment)

---

*[tests/CMakeLists.txt, line 130 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfFnkC40qJcsf1eMIt:-LDfFnkC40qJcsf1eMIu:b2nhrub) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/CMakeLists.txt#L130)):*
> ```CMake
> 
> if(NO_GCC_VARIADIC_TEMPLATE_BUG)
> 	# EX libpmemobj cpp
> ```

XXX

---

*[tests/helpers.cmake, line 155 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfGqOS5_aOz52i8a13:-LDfGqOS5_aOz52i8a14:b3kre) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/helpers.cmake#L155)):*
> ```CMake
>         message(STATUS "Stderr:\n${ERR}")
>     endif()
> endfunction()
> ```

?

---

*[tests/allocator/allocator.cpp, line 1 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfGwIK4rvnkLGRYtvh:-LDfGwIK4rvnkLGRYtvi:b3kqx) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/allocator/allocator.cpp#L1)):*
> ```cpp
> 	/*
> ```

.

---

*[tests/common/pthread_common.hpp, line 42 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfHEFC436tBvb2oPMm:-LDfHEFC436tBvb2oPMn:b-rgtnjh) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/common/pthread_common.hpp#L42)):*
> ```cpp
> 	void *(*start_routine) (void *), void *arg)
> {
> 	if(pthread_create(thread, attr, start_routine, arg) != 0)
> ```

space between if and (

---

*[tests/common/pthread_common.hpp, line 45 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfHIq29KQ74ZDKojmQ:-LDfHIq4BNOzpLgTAxcN:b1r2gsp) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/common/pthread_common.hpp#L45)):*
> ```cpp
> 		UT_FATAL("pthread_create failed");
> 
> 	return 0;
> ```

this function doesn't return anything other than 0, so please convert it to return void

---

*[tests/common/pthread_common.hpp, line 51 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfHGd0D3MISkLJ422O:-LDfHGd0D3MISkLJ422P:b-7ropln) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/common/pthread_common.hpp#L51)):*
> ```cpp
> ut_pthread_join(pthread_t *thread, void **value)
> {
> 	if(pthread_join(*thread, value) != 0)
> ```

ditto

---

*[tests/common/pthread_common.hpp, line 54 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfHQxz1Ex-gUbtn42c:-LDfHQxz1Ex-gUbtn42d:b-7ropln) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/common/pthread_common.hpp#L54)):*
> ```cpp
> 		UT_FATAL("pthread_join failed");
> 
> 	return 0;
> ```

ditto

---

*[tests/common/pthread_common.hpp, line 58 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfHAUf6PsmHA9PeeJi:-LDfHAUf6PsmHA9PeeJj:b3kqx) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/common/pthread_common.hpp#L58)):*
> ```cpp
> 
> #endif // PTHREAD_COMMON_HPP
> ```

.

---

*[tests/common/unittest.hpp, line 53 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfHZTrEwnxz0xLOO4m:-LDfHZTrEwnxz0xLOO4n:b-ytdi97) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/common/unittest.hpp#L53)):*
> ```cpp
> {
>     fprintf(stdout, args...);
>     fprintf(stdout, "\n");
> ```

hmm, i'm wondering if we should use C++ io...

---

*[tests/common/unittest.hpp, line 62 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfHhHt-Fa5Wv8DVir4:-LDfHhHvB_V66Q-uiXWr:bdzfe3m) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/common/unittest.hpp#L62)):*
> ```cpp
>     fprintf(stderr, args...);
>     fprintf(stderr, "\n");
>     exit(1);
> ```

abort, gdb will automatically stop on this line

---

*[tests/common/unittest.hpp, line 91 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfHrOG0GYcSlm5zd4Y:-LDfHrOG0GYcSlm5zd4Z:b3kqx) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/common/unittest.hpp#L91)):*
> ```cpp
> 	(unsigned long long)(lhs), #rhs, (unsigned long long)(rhs)), 0)))
> ```

.

---

*[tests/pool/pool.cpp, line 73 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfJ-Tl6gz9SVX3bnz5:-LDfJ-Tl6gz9SVX3bnz6:b71ula2) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/pool/pool.cpp#L73)):*
> ```cpp
> 	os_stat_t stbuf;
> 	auto ret = os_stat(path, &stbuf);
> 	UT_ASSERTne(ret, -1);
> ```

STAT exists because we didn't want to repeat such checks each time.
I think it's a good idea to retain it in this repo.

---

*[tests/v/v.cpp, line 102 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfK0Ab2_51vAdctzqQ:-LDfK0Ab2_51vAdctzqR:b3kqx) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/v/v.cpp#L102)):*
> ```cpp
> 
> 	pop.close();
> }
> ```

.

---


*Comments from [Reviewable](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2)*
<!-- Sent from Reviewable.io -->

marcinslusarz commented:
Review status: 108 of 238 files reviewed at latest revision, 18 unresolved discussions.

---

*[tests/CMakeLists.txt, line 241 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfOqNW2soADW5B0uwE:-LDfOqNW2soADW5B0uwF:bmn3678) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/CMakeLists.txt#L241)):*
> ```CMake
> 
> # adds all tests within ${name} directory without tracers
> function(add_tests name)
> ```

It's quite confusing to have "add_tests" and "add_test_generic", because it's not clear from the name what is the difference...

For now, I'd delete this one.

---

*[tests/CMakeLists.txt, line 346 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfMS-fAIkRlmj9Bk_9:-LDfMS-fAIkRlmj9Bk_A:b-nd0zlj) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/CMakeLists.txt#L346)):*
> ```CMake
> add_test_generic(timed_mtx_posix drd)
> add_test_generic(timed_mtx_posix helgrind)
> add_test_generic(timed_mtx_posix pmemcheck)
> ```

please sort this list by test name

---

*[tests/common/pthread_common.hpp, line 42 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfNHx42_P7TYfbTlT5:-LDfNHx42_P7TYfbTlT6:bqf1pwp) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/common/pthread_common.hpp#L42)):*
> ```cpp
> 	void *(*start_routine) (void *), void *arg)
> {
> 	if(pthread_create(thread, attr, start_routine, arg) != 0)
> ```

please don't forget to setup code style checker

---


*Comments from [Reviewable](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2)*
<!-- Sent from Reviewable.io -->

marcinslusarz commented:
Review status: 165 of 238 files reviewed at latest revision, 21 unresolved discussions.

---

*[tests/ex_libpmemobj_cpp/Makefile, line 1 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfTq5h32pd0AIhM6Xz:-LDfTq5h32pd0AIhM6Y-:bktxho1) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/437457708ea16fbd2a6bce7fa6e617414c8cac71/tests/ex_libpmemobj_cpp/Makefile#L1)):*
> ```Makefile
> #
> ```

There is no replacement for files in this directory, so I think you should not remove it yet.

---

*[tests/pool/pool_0.cmake, line 39 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfVU407U_3sVvpYQ8f:-LDfVU407U_3sVvpYQ8g:b-7j669e) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/pool/pool_0.cmake#L39)):*
> ```CMake
> execute_with_output("out0.log" ${TEST_EXECUTABLE} c ${DIR}/testfile "test" 20 0600)
> 
> check_file(${DIR}/testfile)
> ```

This function should be renamed to something that suggests what it does.

---

*[tests/pool/pool_0.cmake, line 41 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfVsOA9eWEsfcCdDUV:-LDfVsOA9eWEsfcCdDUW:b-kt2xzc) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/pool/pool_0.cmake#L41)):*
> ```CMake
> check_file(${DIR}/testfile)
> 
> match(out0.log ${SRC_DIR}/out0.log.match)
> ```

This should be automated somehow. May be in the next PR.

---


*Comments from [Reviewable](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2)*
<!-- Sent from Reviewable.io -->

marcinslusarz commented:
Review status: 219 of 238 files reviewed at latest revision, 24 unresolved discussions.

---

*[tests/pool_win/pool_win.cmake, line 32 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfhOGK1SZm92ujVXJ8:-LDfhOGK1SZm92ujVXJ9:b-41lpfk) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/pool_win/pool_win.cmake#L32)):*
> ```CMake
> # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
> 
> include(${SRC_DIR}/../helpers.cmake)
> ```

This test is not wired up.

---

*[tests/ptr_arith/ptr_arith_0.cmake, line 32 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfhpH_CcKV-ufMY22R:-LDfhpH_CcKV-ufMY22S:b91ef4m) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/ptr_arith/ptr_arith_0.cmake#L32)):*
> ```CMake
> # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
> 
> include(${SRC_DIR}/../helpers.cmake)
> ```

Again, not wired up.

---


*Comments from [Reviewable](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2)*
<!-- Sent from Reviewable.io -->

igchor commented:
Review status: 213 of 240 files reviewed at latest revision, 26 unresolved discussions.

---

*[CMakeLists.txt, line 33 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfCnqx8dbswVoyLsno:-LDfi1IY0_35QhdaEUll:b-xhw028) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/CMakeLists.txt#L33)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

C?
</blockquote></details>

Needed to compile check_license.

---

*[CMakeLists.txt, line 87 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfCqPhBB2Wo1_2o_VS:-LDfi6xGB62TRmrs_drb:bss4a2b) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/CMakeLists.txt#L87)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

is this needed?
</blockquote></details>

Removed.

---

*[CMakeLists.txt, line 138 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfDWTS4um6RIunj_kW:-LDfiQmE4uRn09jYx7bs:bss4a2b) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/CMakeLists.txt#L138)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

Is it needed?
</blockquote></details>

Removed.

---

*[tests/CMakeLists.txt, line 102 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfFfXp1WdRwU3Q5gK_:-LDg5t0L4GYv1KEPHl5x:b-ncb0tz) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/CMakeLists.txt#L102)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

file_?
</blockquote></details>

This was taken from pmemfile, but that's indeed not necessary. 

---

*[tests/CMakeLists.txt, line 113 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfFhfb0P8aDxrADDIU:-LDfryrXEfL0VIjf1Gps:b-896fix) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/CMakeLists.txt#L113)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

Why? (please add comment)
</blockquote></details>

Done.

---

*[tests/CMakeLists.txt, line 130 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfFnkC40qJcsf1eMIt:-LDfrzeEEFk-XFb99V7i:b-896fix) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/CMakeLists.txt#L130)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

XXX
</blockquote></details>

Done.

---

*[tests/CMakeLists.txt, line 241 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfOqNW2soADW5B0uwE:-LDfs1iw32c8CukRFy6z:brs95z5) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/CMakeLists.txt#L241)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

It's quite confusing to have "add_tests" and "add_test_generic", because it's not clear from the name what is the difference...

For now, I'd delete this one.
</blockquote></details>

This function adds all tests from one directory (all .cmake files). add_test_generic only adds one test (which can be specified, default is first test  - test_0.cmake). If we would remove add_tests we would need to manually add all tests, e.g.:
add_test_generic(some_test 0)
add_test_generic(some_test 1)
...

---

*[tests/CMakeLists.txt, line 346 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfMS-fAIkRlmj9Bk_9:-LDfxz-77I3-8M1yCMp6:b-896fix) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/CMakeLists.txt#L346)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

please sort this list by test name
</blockquote></details>

Done.

---

*[tests/helpers.cmake, line 155 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfGqOS5_aOz52i8a13:-LDfy9M9EEKnbUEhMmZ4:b-896fix) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/helpers.cmake#L155)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

?
</blockquote></details>

Done.

---

*[tests/allocator/allocator.cpp, line 1 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfGwIK4rvnkLGRYtvh:-LDfyISV2v1pAujcVzXq:b-896fix) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/allocator/allocator.cpp#L1)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

.
</blockquote></details>

Done.

---

*[tests/common/pthread_common.hpp, line 42 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfHEFC436tBvb2oPMm:-LDfyJPD4ZFzFDhHhz2u:b-896fix) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/common/pthread_common.hpp#L42)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

space between if and (
</blockquote></details>

Done.

---

*[tests/common/pthread_common.hpp, line 42 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfNHx42_P7TYfbTlT5:-LDgCs4c2ybVaoQn_Vnu:b-896fix) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/common/pthread_common.hpp#L42)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

please don't forget to setup code style checker
</blockquote></details>

Done.

---

*[tests/common/pthread_common.hpp, line 45 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfHIq29KQ74ZDKojmQ:-LDfyRVc7gEQks9fNReg:b-896fix) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/common/pthread_common.hpp#L45)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

this function doesn't return anything other than 0, so please convert it to return void
</blockquote></details>

Done.

---

*[tests/common/pthread_common.hpp, line 51 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfHGd0D3MISkLJ422O:-LDfyR97CsFt6DIVKVSR:b-896fix) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/common/pthread_common.hpp#L51)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

Done.

---

*[tests/common/pthread_common.hpp, line 54 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfHQxz1Ex-gUbtn42c:-LDfyRuGCo35u5qcHcEQ:b-896fix) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/common/pthread_common.hpp#L54)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

Done.

---

*[tests/common/pthread_common.hpp, line 58 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfHAUf6PsmHA9PeeJi:-LDg16DICPqqIOEbqaU8:b-896fix) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/common/pthread_common.hpp#L58)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

.
</blockquote></details>

Done.

---

*[tests/common/unittest.hpp, line 53 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfHZTrEwnxz0xLOO4m:-LDg--ASCiuW0wGzqSyW:bd5fj8g) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/common/unittest.hpp#L53)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

hmm, i'm wondering if we should use C++ io...
</blockquote></details>

Ok, this can be changed in another PR.

---

*[tests/common/unittest.hpp, line 62 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfHhHt-Fa5Wv8DVir4:-LDfzzs9CP9cY8-tTEkU:b-896fix) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/common/unittest.hpp#L62)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

abort, gdb will automatically stop on this line
</blockquote></details>

Done.

---

*[tests/common/unittest.hpp, line 91 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfHrOG0GYcSlm5zd4Y:-LDg-DX_6d5OMCwdAaXE:b-896fix) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/common/unittest.hpp#L91)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

.
</blockquote></details>

Done.

---

*[tests/pool/pool.cpp, line 73 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfJ-Tl6gz9SVX3bnz5:-LDg2Lu-7MOGg643xLO9:b-cd8szy) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/pool/pool.cpp#L73)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

STAT exists because we didn't want to repeat such checks each time.
I think it's a good idea to retain it in this repo.
</blockquote></details>

It is actually only used once - in this place (there is also one use of STATW - in pool_win.cpp). Is it worth declaring macro?

---

*[tests/v/v.cpp, line 102 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfK0Ab2_51vAdctzqQ:-LDg5Xbg4ww1yXzDhYFI:b-896fix) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/v/v.cpp#L102)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

.
</blockquote></details>

Done.

---

*[tests/ex_libpmemobj_cpp/Makefile, line 1 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfTq5h32pd0AIhM6Xz:-LDg2Jj13oRo7xqHz8Rv:b-896fix) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/437457708ea16fbd2a6bce7fa6e617414c8cac71/tests/ex_libpmemobj_cpp/Makefile#L1)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

There is no replacement for files in this directory, so I think you should not remove it yet.
</blockquote></details>

Done.

---

*[tests/pool/pool_0.cmake, line 39 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfVU407U_3sVvpYQ8f:-LDg4dK56k9HBF7nRVtz:b-896fix) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/pool/pool_0.cmake#L39)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

This function should be renamed to something that suggests what it does.
</blockquote></details>

Done.

---

*[tests/pool/pool_0.cmake, line 41 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfVsOA9eWEsfcCdDUV:-LDg4e9t195TU9pixMuE:b5a4lvs) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/pool/pool_0.cmake#L41)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

This should be automated somehow. May be in the next PR.
</blockquote></details>

Ok, however there are not many tests with match files.

---

*[tests/ptr_arith/ptr_arith_0.cmake, line 32 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfhpH_CcKV-ufMY22R:-LDg5P8YA7gB8GAsYooc:b-896fix) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/ptr_arith/ptr_arith_0.cmake#L32)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

Again, not wired up.
</blockquote></details>

Done.

---

*[tests/pool_win/pool_win.cmake, line 32 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfhOGK1SZm92ujVXJ8:-LDg5EBgBkqtCKZi5SEV:bv8jfyn) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/pool_win/pool_win.cmake#L32)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

This test is not wired up.
</blockquote></details>

It will be added in some next PR.

---


*Comments from [Reviewable](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2)*
<!-- Sent from Reviewable.io -->

marcinslusarz commented:
Review status: 214 of 240 files reviewed at latest revision, 3 unresolved discussions.

---

*[tests/CMakeLists.txt, line 241 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfOqNW2soADW5B0uwE:-LDgG0yO3EbDTII2NOjA:bjzq00q) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/CMakeLists.txt#L241)):*
<details><summary><i>Previously, igchor wrote…</i></summary><blockquote>

This function adds all tests from one directory (all .cmake files). add_test_generic only adds one test (which can be specified, default is first test  - test_0.cmake). If we would remove add_tests we would need to manually add all tests, e.g.:
add_test_generic(some_test 0)
add_test_generic(some_test 1)
...
</blockquote></details>

I know what it does. And I think it makes sense to list all tests here. Otherwise GLOB may match files which were not committed yet.
It's also easier to manage which test runs with which tracer.

---

*[tests/CMakeLists.txt, line 346 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfMS-fAIkRlmj9Bk_9:-LDgFDGU4zOjQbIHn-e_:b6gf6xc) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/CMakeLists.txt#L346)):*
<details><summary><i>Previously, igchor wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Q is between P and R :)

---


*Comments from [Reviewable](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2)*
<!-- Sent from Reviewable.io -->

igchor commented:
Review status: 239 of 240 files reviewed at latest revision, 2 unresolved discussions.

---

*[tests/CMakeLists.txt, line 241 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfOqNW2soADW5B0uwE:-LDgX6MoBENd02Rp1vM_:bl1h72v) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/CMakeLists.txt#L241)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

I know what it does. And I think it makes sense to list all tests here. Otherwise GLOB may match files which were not committed yet.
It's also easier to manage which test runs with which tracer.
</blockquote></details>

Ok, done.

---

*[tests/CMakeLists.txt, line 346 at r1](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2#-LDfMS-fAIkRlmj9Bk_9:-LDgX3b5CTrEYF3_Vthu:b-896fix) ([raw file](https://github.com/pmem/libpmemobj-cpp/blob/700bac516a122831f0aa08be4a7269812d501393/tests/CMakeLists.txt#L346)):*
<details><summary><i>Previously, marcinslusarz (Marcin Ślusarz) wrote…</i></summary><blockquote>

Q is between P and R :)
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io/reviews/pmem/libpmemobj-cpp/2)*
<!-- Sent from Reviewable.io -->
